GeosCore/diagnostics_mod.F90

```diff
@@ -1301,6 +1311,37 @@ SUBROUTINE Compute_Budget_Diagnostics( Input_Opt,   State_Chm, State_Grid, &
     ENDDO
     !$OMP END PARALLEL DO
 
+    ! Budget consistency check
+    ! The global total tracer mass should *not* change between an "after" call and the next "before" call
+    if ( before .and. (.not. any( PriorGlobalMass == 0. )) ) then
+       
+       ! Relative error, change since prior "after" call
+       rerr(1) = abs( sum( colMass(:,:,:,1) ) / PriorGlobalMass(1) - 1 )
+       rerr(2) = abs( sum( colMass(:,:,:,2) ) / PriorGlobalMass(2) - 1 )
+       rerr(3) = abs( sum( colMass(:,:,:,3) ) / PriorGlobalMass(3) - 1 )
+       
+       ! Relative error tolarance, errors larger than this will halt the model
+       rtol = 0.001
+
+       IF ( rerr(1) > rtol ) THEN
+         print*,'Budget Diagnostic Error: FULL-column tracer mass changed unexpectedly, rerr=',rerr(1)
+       ENDIF
+       IF ( rerr(2) > rtol ) THEN
+         print*,'Budget Diagnostic Error: TROP-column tracer mass changed unexpectedly, rerr=', rerr(2)
```

Comment:
All of these prints need to be wrapper in Input_Opt%amIRoot logical to avoid prints from all cores when using MPI.